controllers/backplaneconfig_controller.go

```diff
@@ -312,11 +322,62 @@ func (r *MultiClusterEngineReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return result, err
 	}
 
+	if upgrade {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
 	r.StatusManager.AddCondition(status.NewCondition(backplanev1.MultiClusterEngineProgressing, metav1.ConditionTrue, status.DeploySuccessReason, "All components deployed"))
 
 	return ctrl.Result{}, nil
 }
 
+func (r *MultiClusterEngineReconciler) setOperatorUpgradeableStatus(ctx context.Context, m *backplanev1.MultiClusterEngine) (bool, error) {
+
+	// Temporary variable
+	var upgradeable bool
+
+	// Checking to see if the current version of the MCE matches the desired to determine if we are in an upgrade scenario
+	// If the current version doesn't exist, we are currently in a install which will also not allow it to upgrade
+
+	if m.Status.CurrentVersion != m.Status.DesiredVersion {
```

Comment:
We don't want to peer at our own status if we don't need to. In this case `m.Status.DesiredVersion` can be replaced with `version.Version`. Otherwise we are looking at what the desired version _was_ last time we updated status.